src/main/asciidoc/_chapters/developer.adoc

```diff
@@ -420,6 +419,27 @@ Tests may not all pass so you may need to pass `-DskipTests` unless you are incl
 
 To pick a particular Hadoop 3.y release, you'd set hadoop-three.version property e.g. `-Dhadoop-three.version=3.0.0`.
 
+[[maven.build.jdk11_hadoop3]]
+==== Building with JDK11 and Hadoop3
+
+The following guidance applies to HBase versions [2.3, 3), where both JDK8 and JDK11 are supported,
+and both Hadoop2 and Hadoop3 are supported.
+
+HBase manages JDK-specific build settings using maven profiles. The profile appropriate to the JDK
+in use is automatically selected. JDK8 supports both Hadoop2 and Hadoop3. For JDK11, only Hadoop3
+is supported, which means you must specify `-Dhadoop.profile=3.0` when building with JDK11.
```

Comment:
Is this really necessary as the `hadoop-3.0` profile is active by default (at least on master)? Or better state it to make it generic for all mentioned releases?

Reply:
By specifying `-Dhadoop.profile=3.0`, users are consistent across all JDK 11 supported branches. For instance, branch-2 has `hadoop-2.0` active by default I believe. Hence, with this doc, users can try `-Dhadoop.profile=3.0` for all supported branches and there is no harm with master branch also. This should be fine. Thought?

Reply:
That's what I meant by the second question. For me it's totally fine. Just wanted to shortly discuss if we should make a distinction between the different releases or not.

Reply:
Oh okk, got it. Thanks

Reply:
I called out that this is applicable to 2.3, up through but excluding 3.0. You think having this documented here will be confusing? Should I point out how 3.0 is different?

Reply:
My intention (and I believe it's common practice for all release managers) is to make a blanket-copy of the docs from master back to branch-2 and branch-2.3. I've staged the commit locally and skimming through, I haven't found much content that's applicable to master but not branch-2/.3.

Reply:
I think the earlier section, [Building against various hadoop versions](http://hbase.apache.org/book.html#maven.build.hadoop) needs updated in light of the point you've raised here. Nice find.

Reply:
I've updated that text as well, let me know if it satisfies.

Reply:
Bah. Now I need to update this text to refer back to what I just wrote.